pkg/onit/cli/cli.go

```diff
@@ -1121,20 +1123,66 @@ func getDebugCommand() *cobra.Command {
 			if err != nil {
 				exitError(err)
 			}
-
 			port, _ := cmd.Flags().GetInt("port")
-			if err := cluster.PortForward(args[0], port, 40000); err != nil {
+			nodes, err := cluster.GetNodes()
+			if err != nil {
 				exitError(err)
+			}
+
+			if len(args) == 0 {
+				var wg sync.WaitGroup
+				n := len(nodes)
+				wg.Add(n)
+
+				asyncErrors := make(chan error, n)
+				debugPort := onit.DebugPort
+				checked := make(chan bool, 1)
+				var stat bool
+
+				for index := range nodes {
+					onit.CheckFreePort(int(debugPort), checked)
+					stat = <-checked
+					for !stat {
+						debugPort = debugPort + 1
+						onit.CheckFreePort(int(debugPort), checked)
+						stat = <-checked
+
+					}
+					go func(node onit.NodeInfo, port int) {
+						fmt.Println("Start Debugger for:", node.ID)
+						err = cluster.PortForward(node.ID, int(debugPort), 40000)
+						asyncErrors <- err
+						wg.Done()
+					}(nodes[index], int(debugPort))
+
+					time.Sleep(100 * time.Millisecond)
```

Comment:
I think if we just increment `debugPort` here to ensure the next iteration doesn't check the same port, we shouldn't need to sleep to wait for the port to be bound. There's always going to be a race between checking the availability of the port and finally binding to it, so we should just fail fast (via `asyncErrors`) if a port is bound by the time we try to bind to it.